src/java/org/apache/cassandra/db/marshal/CompositeType.java

```diff
@@ -142,6 +144,14 @@ protected CompositeType(List<AbstractType<?>> types)
         this.types = ImmutableList.copyOf(types);
     }
 
+    @Override
+    public CompositeType overrideKeyspace(Function<String, String> overrideKeyspace)
+    {
+        return new CompositeType(subTypes().stream()
```

Comment:
see https://github.com/datastax/cassandra/pull/331/files#r801303769, we should use `getInstance` to reduce GC.